syntax = "proto3";

import "google/protobuf/empty.proto";

package experiment;

service DBGatewayService {
    rpc QueryOutOfRange(QueryOutOfRangeRequest) returns (QueryOutOfRangeResponse);
    rpc QueryTemperature(QueryTemperatureRequest) returns (QueryTemperatureResponse);
    rpc AddTemperature(AddTemperatureRequest) returns (google.protobuf.Empty);
}

message TemperatureRecord {
    double timestamp = 1;
    float temperature = 2;
}

message QueryOutOfRangeRequest {
    string experiment_id = 1;
}

message QueryOutOfRangeResponse {
    repeated TemperatureRecord temperatures = 1;
}

message QueryTemperatureRequest {
    string experiment_id = 1;
    double start_epoch_time = 2;
    double end_epoch_time = 3;
}

message QueryTemperatureResponse {
    repeated TemperatureRecord temperatures = 1;
}

message AddTemperatureRequest {
    string experiment_id = 1;
    bool out_of_bounds = 2;
    double timestamp = 3;
    float temperature = 4;
}
